document/src/vespa/document/base/idstring.cpp

```diff
@@ -153,8 +153,13 @@ inline const char *
 fmemchr(const char * s, const char * e)
 {
     while (s+15 < e) {
+#ifdef __clang__
+        v16qi tmpCurrent = __builtin_ia32_lddqu(s);
+        v16qi tmp0       = tmpCurrent == _G_zero;
+#else
         v16qi tmpCurrent = __builtin_ia32_loaddqu(s);
         v16qi tmp0       = __builtin_ia32_pcmpeqb128(tmpCurrent, _G_zero);
+#endif
```

Comment:
We should rewrite this later to use more standardized intrinsics such as `_mm_loadu_si128`, `_mm_movemask_epi8` etc. I did that locally at some point when running the doc id parser code through libFuzzer, will try to dig it up.

Reply:
Do not forget cleanup.